src/librustc_mir/util/liveness.rs

```diff
@@ -281,24 +283,25 @@ pub fn categorize<'tcx>(context: PlaceContext<'tcx>, mode: LivenessMode) -> Opti
     }
 }
 
-struct DefsUsesVisitor {
+struct DefsUsesVisitor<V: Idx> {
     mode: LivenessMode,
-    defs_uses: DefsUses,
+    defs_uses: DefsUses<V>,
 }
 
 #[derive(Eq, PartialEq, Clone)]
-struct DefsUses {
-    defs: LocalSet,
-    uses: LocalSet,
+struct DefsUses<V: Idx>
```

Comment:
This struct is going to need to take the `&impl LiveVariableMap` and store it in a field (`map`, say)